pkg/discovery/dns/provider.go

```diff
@@ -91,6 +97,7 @@ func (p *Provider) Resolve(ctx context.Context, addrs []string) {
 		p.resolverLookupsCount.Inc()
 		if err != nil {
 			// The DNS resolution failed. Continue without modifying the old records.
+			p.resolved[addr] = p.resolved[addr] // Ensure metrics capture the result even if empty.
```

Comment:
I don't really get this part is, it intentional?  Isn't this noop?
It's tricky what should be there TBH ;p Should we set results for both query to empty and metrics to 0? should we only adjust metrics? Not sure yet.

*

Reply:
This ensures that the map contains an entry for the address. So when we iterate over the map keys for the metric, we report that there was an address configured regardless of whether we got some results here, got no results, or continue using previous results.